src/middleware/rewriteUrls.coffee

```diff
@@ -0,0 +1,104 @@
+url = require 'url'
+winston = require 'winston'
+Q = require 'q'
+
+Channel = (require '../../lib/model/channels').Channel
+config = require '../config/config'
+routerConf = config.get 'router'
+utils = require '../../lib/utils'
+router = require '../../lib/middleware/router'
+
+invertPathTransform = (pathTransform) ->
+  return pathTransform.replace /s\/(.*?)\/(.*?)($|\/(.*)$)/, 's/$2/$1/$4'
+
+exports.fetchRewriteConfig = (channel, authType, callback) ->
+  # set rewrite config from current channel
+  rwConfig = []
+  if channel.rewriteUrlsConfig?
+    rwConfig = channel.rewriteUrlsConfig
+
+  # add in default rewrite rules for hosts we proxy
+  utils.getAllChannels (err, channels) ->
+    if err?
+      return callback err
+
+    for channel in channels
+      for route in channel.routes
+        do ->
+          if route.primary
+            if route.pathTransform
+              inverveTransform = invertPathTransform route.pathTransform
+
+            if authType? is 'tls'
+              toPort = routerConf.httpsPort
+            else
+              toPort = routerConf.httpPort
+
+            rwConfig.push
+              'fromHost':       route.host
+              'toHost':         routerConf.externalHostname
+              'fromPort':       route.port
+              'toPort':         toPort
+              'pathTransform':  if inverveTransform then inverveTransform else null
+
+    callback null, rwConfig
+
+rewriteUrls = (body, channel, authType, callback) ->
+  exports.fetchRewriteConfig channel, authType, (err, rwConfig) ->
+    if err?
+      return callback err
+
+    # rewrite each found href attribute (in JSON or XML)
+    newBody = body.replace /["|']?href["|']?[:|=]\s?["|'](\S*?)["|']/g, (match, hrefUrl) ->
```

Comment:
Maybe we should check for `src=` as well? Just looking at a CSD page for reference. In fact it might be good just to test against CSD to see that this works for it, because, I believe, this issue was largely motivated by it?